products/BellHybrid/apps/application-bell-focus-timer/data/FocusContract.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <apps-common/BasePresenter.hpp>
+#include <widgets/ListItemProvider.hpp>
+#include <apps-common/widgets/TimerWithCallbacks.hpp>
+#include <time/time_locale.hpp>
+
+namespace app::focus::contract
+{
+    class View
+    {
+      public:
+        virtual ~View() noexcept = default;
+    };
+
+    class SettingsPresenter : public BasePresenter<View>
+    {
+      public:
+        virtual ~SettingsPresenter() noexcept                                           = default;
+        virtual auto getPagesProvider() const -> std::shared_ptr<gui::ListItemProvider> = 0;
+        virtual void loadData()                                                         = 0;
+        virtual void saveData()                                                         = 0;
+        virtual void eraseProviderData()                                                = 0;
+        virtual void exitWithSave()                                                     = 0;
+        virtual void exitWithoutSave()                                                  = 0;
+    };
+
+    class TimerPresenter : public BasePresenter<View>
+    {
+      public:
+        virtual ~TimerPresenter() noexcept                                      = default;
+        virtual void setTimer(std::unique_ptr<app::TimerWithCallbacks> &&timer) = 0;
+        virtual void handleUpdateTimeEvent()                                    = 0;
+        virtual bool isTimerStopped()                                           = 0;
+        virtual void start()                                                    = 0;
+        virtual void stop()                                                     = 0;
+        virtual void pause()                                                    = 0;
+        virtual void resume()                                                   = 0;
+        virtual void abandon()                                                  = 0;
+        virtual void finish()                                                   = 0;
+        virtual void onBeforeShow()                                             = 0;
+    };
+} // namespace app::focus::contract
```

Comment:
is this file needed? I see there are separate contracts for dedicated presenters.